clients/python/treedb_client/src/treedb_client/client.py

```
@@ -483,9 +545,16 @@ def _normalize_base_url(base_url: str) -> str:
     if not isinstance(base_url, str) or not base_url.strip():
         raise TreeDBConfigError("base_url must be a non-empty HTTP(S) URL")
     trimmed = base_url.strip().rstrip("/")
-    parsed = urllib.parse.urlparse(trimmed)
+    try:
+        parsed = urllib.parse.urlparse(trimmed)
+        hostname = parsed.hostname
+        parsed.port
```

Ruff (0.16.1)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)